src/main/java/graphql/nadel/schema/NadelDirectives.java

```diff
@@ -0,0 +1,269 @@
+package graphql.nadel.schema;
+
+import graphql.Assert;
+import graphql.language.Description;
+import graphql.language.DirectiveDefinition;
+import graphql.language.InputObjectTypeDefinition;
+import graphql.language.IntValue;
+import graphql.language.NonNullType;
+import graphql.language.SourceLocation;
+import graphql.language.StringValue;
+import graphql.nadel.dsl.FieldMappingDefinition;
+import graphql.nadel.dsl.RemoteArgumentDefinition;
+import graphql.nadel.dsl.RemoteArgumentSource;
+import graphql.nadel.dsl.TypeMappingDefinition;
+import graphql.nadel.dsl.UnderlyingServiceHydration;
+import graphql.schema.GraphQLArgument;
+import graphql.schema.GraphQLDirective;
+import graphql.schema.GraphQLDirectiveContainer;
+import graphql.schema.GraphQLFieldDefinition;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static graphql.Assert.assertNotNull;
+import static graphql.Assert.assertTrue;
+import static graphql.introspection.Introspection.DirectiveLocation.ENUM;
+import static graphql.introspection.Introspection.DirectiveLocation.ENUM_VALUE;
+import static graphql.introspection.Introspection.DirectiveLocation.FIELD_DEFINITION;
+import static graphql.introspection.Introspection.DirectiveLocation.INPUT_OBJECT;
+import static graphql.introspection.Introspection.DirectiveLocation.INTERFACE;
+import static graphql.introspection.Introspection.DirectiveLocation.OBJECT;
+import static graphql.introspection.Introspection.DirectiveLocation.SCALAR;
+import static graphql.introspection.Introspection.DirectiveLocation.UNION;
+import static graphql.language.DirectiveLocation.newDirectiveLocation;
+import static graphql.language.InputValueDefinition.newInputValueDefinition;
+import static graphql.language.ListType.newListType;
+import static graphql.language.TypeName.newTypeName;
+import static graphql.nadel.dsl.RemoteArgumentSource.SourceType.CONTEXT;
+import static graphql.nadel.dsl.RemoteArgumentSource.SourceType.FIELD_ARGUMENT;
+import static graphql.nadel.dsl.RemoteArgumentSource.SourceType.OBJECT_FIELD;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+
+public class NadelDirectives {
+
+    static final DirectiveDefinition HIDE_DIRECTIVE_DEFINITION;
```

Comment:
we should think about this naming for a second and make sure we consider directives for both use case:
directives for the overall schema (when you still have an underlying one) and the directives if you only have one underlying schema.

Reply:
can we remove the hide directive for now and revisit it again when we need it?

Reply:
removed